app/components/course-page/course-stage-step/community-solution-card/content.hbs

```diff
@@ -54,9 +54,6 @@
   <FileContentsCard
     @headerTooltipText="This file was added in previous stages."
     @isCollapsible={{true}}
-    @isCollapsed={{not (includes unchangedFileComparison.path this.expandedUnchangedFilePaths)}}
```

Comment:
We still need this functionality @VasylMarchuk, don't think it can be removed. For testing this, you should be able to load any "view community solution" / "view code example" test

Reply:
The functionality didn't go anywhere, I just moved `@isCollapsed` into an internal component's property `isCollapsed` and made the component itself responsible for collapsing / expanding, so parent components don't have to do it. If we need to know the current state — we can `yield this.isCollapsed`.

If we need to really know which files are collapsed (to add them to some kind of array) — I can add `@onCollapse/@onExpand`  arguments back — but nothing is currently using it, I checked thoroughly.

Also, nothing currently passes files to be expanded or collapsed initially, but if needed, I can add `@initiallyExpanded` argument.

Reply:
In this case it isn't just knowing the current state, we might need to also trigger collapse/expand from outside the component - I see this as a use-case for a "controlled" property, not an uncontrolled property.

Example of things we'll use this for in the future: 

- Fire analytics events when users expand/collapse files using `@onCollapse`/`@onExpand`
- Render a list of files where only one file can be expanded at a time (i.e. we automatically collapse other files when a user clicks on a file). 

I don't think that all of this will be possible if `@isCollapsed` is an uncontrolled property (let me know if you think otherwise though)

Reply:
I guess you're right, this was bothering me a lot, since collapse/expand/isExpanded logic needs to be duplicated in every parent component. But I intentionally did this change in a separate commit, so it's easily undone, I'll just undo it for now.